### DISCUSSION TITLE
How to use `cork` properly?

### DISCUSSION THREAD

**Discussion by dalisoft** [uNetworking/uWebSockets.js#1092] [Q&A] [answered]
Hi everyone

I am building some more complex application which helps create backend websites easier without sacrificing performance (all notes and credits about uWebSockets.js will be on app).

But there i have to ask question about `HttpResponse.cork`

## Separate cork

```js
app.get('/cork', async (res) => {
  res.cork(() => {
    res.writeStatus('301 Moved Permanently');
  });
  res.cork(() => {
    res.writeHeader('Content-Type', 'application/text');
  });
  res.cork(() => {
    res.end('success');
  });
});
```

## Single cork

```js
app.get('/cork', async (res) => {
  res.cork(() => {
    res.writeStatus('301 Moved Permanently');
    res.writeHeader('Content-Type', 'application/text');
    res.end('success');
  });
});
```

I know **Single cork** will be faster and more efficiently but it is possible to use **Separate cork** way and it is not bad / bad practice? 
Because my application requires **Separate cork**, using `array of handlers` also not possible due of architecture of my application

If i use **Separate cork** way, **uWebSockets.js** will merge those into single **cork** automatically or not?

**Comment by dalisoft** [comment, marked as answer]
Used another approach and it's more efficient in terms of response but performance may be not the fastest

### Batch cork

```js
app.get('/cork', async (res) => {
  const corks = [];
  
  corks.push(() => {
    res.writeStatus('301 Moved Permanently');
  });
  corks.push(() => {
    res.writeHeader('Content-Type', 'application/text');
  });
  corks.push(() => {
    res.end('success');
  });
  
  return res.cork(() => {
    for (const cork of corks) {
      cork();
    }
  });
});
```

**Comment by uNetworkingAB** [comment]
> all notes and credits about uWebSockets.js will be on app

There is nothing in the license encouraging or requiring you to give any kind of credit. In fact, the license only makes requirements on the opposite.

Separate cork: horrible, just horrible. Can't make it any worse than that.
Single cork: Good
Batch cork: Not bad, but not ideal. But it does at least actually cork the response

**Reply by dalisoft** [reply]
Thank you for response. 
I mean give add note it’s based on uWebSockets.js because without uWebSockets.js my application was not possible
And thank you for your project too
